### PR TITLE
Fix auto-tag build scripts

### PR DESCRIPTION
**Pull request by RedbackThomson**
Description of changes:
When adding the [Go version and architecture to the code-generator Dockerfile](https://github.com/aws-controllers-k8s/code-generator/pull/391), it broke the continuous deployment scripts which use a separate set of calls (to `buildah`) instead of using `build-controller-image.sh`. This pull request adds the same `GOARCH` and `GOLANG_VERSION` build arguments to the `release-controller.sh` script and adds the Go CLI to the deploy container to support them.

These changes were validated by running a custom edit of the Prowjob against the new images and script. Log output: https://prow.ack.aws.dev/view/s3/ack-prow-logs/logs/lambda-post-submit/1620583281521070080

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by RedbackThomson** [comment]
I have tested these changes on the failing Lambda release job. Final output, when using this branch: https://prow.ack.aws.dev/log?job=lambda-post-submit&id=1620583281521070080

**Review by A-Hilaly — APPROVED** [review]
Grazie mille!

File: cd/scripts/release-controller.sh [hunk 1]
```diff
@@ -31,6 +31,7 @@ Environment variables:
 # find out the service name and semver tag from the prow environment variables.
 AWS_SERVICE=$(echo "$REPO_NAME" | rev | cut -d"-" -f2- | rev | tr '[:upper:]' '[:lower:]')
 VERSION=$PULL_BASE_REF
+GOARCH=${GOARCH:-"amd64"}
```
Comment (A-Hilaly) [inline cd/scripts/release-controller.sh:34]: Can we use $(go env GOARCH) instead?

**Review by jljaco — APPROVED** [review]
Looks good

Can you please link in the PR description to the previous PR which broke the scripts?

Also, how will this fix be validated?

**Pull request merged**